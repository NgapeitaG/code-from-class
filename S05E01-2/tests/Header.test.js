import React from 'react'
import { shallow } from 'enzyme'

import Header from '../client/components/Header'

test('<Header /> matches the last snapshot', () => {
  expect(shallow(<Header />)).toMatchSnapshot()
})
