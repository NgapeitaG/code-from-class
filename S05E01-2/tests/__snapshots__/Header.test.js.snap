// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> matches the last snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="brand-title"
        >
          Charlotte’s Web Log
        </h1>,
        <nav
          className="nav"
        >
          <ul
            className="nav-list"
          >
            <li
              className="nav-item"
            >
              <Link
                className="pure-button"
                replace={false}
                to="/"
              >
                Home
              </Link>
            </li>
            <li
              className="nav-item"
            >
              <Link
                className="pure-button"
                replace={false}
                to="/posts/new"
              >
                Add Post
              </Link>
            </li>
          </ul>
        </nav>,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Charlotte’s Web Log",
          "className": "brand-title",
        },
        "ref": null,
        "rendered": "Charlotte’s Web Log",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ul
            className="nav-list"
          >
            <li
              className="nav-item"
            >
              <Link
                className="pure-button"
                replace={false}
                to="/"
              >
                Home
              </Link>
            </li>
            <li
              className="nav-item"
            >
              <Link
                className="pure-button"
                replace={false}
                to="/posts/new"
              >
                Add Post
              </Link>
            </li>
          </ul>,
          "className": "nav",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="nav-item"
              >
                <Link
                  className="pure-button"
                  replace={false}
                  to="/"
                >
                  Home
                </Link>
              </li>,
              <li
                className="nav-item"
              >
                <Link
                  className="pure-button"
                  replace={false}
                  to="/posts/new"
                >
                  Add Post
                </Link>
              </li>,
            ],
            "className": "nav-list",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  className="pure-button"
                  replace={false}
                  to="/"
                >
                  Home
                </Link>,
                "className": "nav-item",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Home",
                  "className": "pure-button",
                  "replace": false,
                  "to": "/",
                },
                "ref": null,
                "rendered": "Home",
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  className="pure-button"
                  replace={false}
                  to="/posts/new"
                >
                  Add Post
                </Link>,
                "className": "nav-item",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Add Post",
                  "className": "pure-button",
                  "replace": false,
                  "to": "/posts/new",
                },
                "ref": null,
                "rendered": "Add Post",
                "type": [Function],
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
        "type": "nav",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="brand-title"
          >
            Charlotte’s Web Log
          </h1>,
          <nav
            className="nav"
          >
            <ul
              className="nav-list"
            >
              <li
                className="nav-item"
              >
                <Link
                  className="pure-button"
                  replace={false}
                  to="/"
                >
                  Home
                </Link>
              </li>
              <li
                className="nav-item"
              >
                <Link
                  className="pure-button"
                  replace={false}
                  to="/posts/new"
                >
                  Add Post
                </Link>
              </li>
            </ul>
          </nav>,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Charlotte’s Web Log",
            "className": "brand-title",
          },
          "ref": null,
          "rendered": "Charlotte’s Web Log",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ul
              className="nav-list"
            >
              <li
                className="nav-item"
              >
                <Link
                  className="pure-button"
                  replace={false}
                  to="/"
                >
                  Home
                </Link>
              </li>
              <li
                className="nav-item"
              >
                <Link
                  className="pure-button"
                  replace={false}
                  to="/posts/new"
                >
                  Add Post
                </Link>
              </li>
            </ul>,
            "className": "nav",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li
                  className="nav-item"
                >
                  <Link
                    className="pure-button"
                    replace={false}
                    to="/"
                  >
                    Home
                  </Link>
                </li>,
                <li
                  className="nav-item"
                >
                  <Link
                    className="pure-button"
                    replace={false}
                    to="/posts/new"
                  >
                    Add Post
                  </Link>
                </li>,
              ],
              "className": "nav-list",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    className="pure-button"
                    replace={false}
                    to="/"
                  >
                    Home
                  </Link>,
                  "className": "nav-item",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Home",
                    "className": "pure-button",
                    "replace": false,
                    "to": "/",
                  },
                  "ref": null,
                  "rendered": "Home",
                  "type": [Function],
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    className="pure-button"
                    replace={false}
                    to="/posts/new"
                  >
                    Add Post
                  </Link>,
                  "className": "nav-item",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Add Post",
                    "className": "pure-button",
                    "replace": false,
                    "to": "/posts/new",
                  },
                  "ref": null,
                  "rendered": "Add Post",
                  "type": [Function],
                },
                "type": "li",
              },
            ],
            "type": "ul",
          },
          "type": "nav",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
